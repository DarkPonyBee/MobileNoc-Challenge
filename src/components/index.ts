export * from './TaskItem';
export * from './Divider';
