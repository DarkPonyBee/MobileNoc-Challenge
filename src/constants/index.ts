export const COUNTER_CHANGE = 'COUNTER_CHANGE';
export const FETCH_TASKS = 'FETCH_TASKS';
export const FETCH_FAILED = 'FETCH_FAILED';

export const Colors = {
  aliceBlue: '#F5FCFF',
  darkCharcoal: '#333333',
  darkBlue: '#124c81',
  darkPurple: '#3c345c',
  darkOragne: '#d6542c',
  darkWhite: 'rgba(50, 50, 50, 0.1)',
};
