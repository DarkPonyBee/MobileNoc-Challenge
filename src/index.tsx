import React from 'react';
import {SafeAreaView, StatusBar} from 'react-native';
import {Provider} from 'react-redux';
import Home from './screens/Home';
import {store} from './store/configureStore';

declare const global: {HermesInternal: null | {}};

const App = () => {
  return (
    <Provider store={store}>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView />
      <Test />
    </Provider>
  );
};

export default App;

const Test = () => {
  return (
    <>
      <Home />
    </>
  );
};
