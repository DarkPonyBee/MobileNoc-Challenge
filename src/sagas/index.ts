import {all} from 'redux-saga/effects';
import tasks from './tasksSagas';

export default function* root() {
  yield all([tasks()]);
}
