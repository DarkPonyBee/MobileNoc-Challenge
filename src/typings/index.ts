export enum TASK_STATUS {
  DONE = 'done',
  PROGRESS = 'in-progress',
  PENDING = 'pending',
}

export type Task = {
  id: string;
  name: string;
  due_date: string;
  status: TASK_STATUS;
};
