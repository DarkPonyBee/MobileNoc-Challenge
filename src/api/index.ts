import {Task} from '../typings';

export default class JSONPlacholderAPI {
  static listApiEntry = 'https://mocky.io/v2/';

  static async fetchTasks(userId: string): Promise<Task[]> {
    const url = JSONPlacholderAPI.listApiEntry + `${userId}`;
    const res = await fetch(url);
    const {tasks} = await res.json();
    return tasks;
  }
}
