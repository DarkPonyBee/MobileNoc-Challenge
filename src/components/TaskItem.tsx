import React, {useState} from 'react';
import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  TouchableOpacity,
} from 'react-native';
import CheckBox from '@react-native-community/checkbox';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import EntypoIcons from 'react-native-vector-icons/Entypo';
import {TASK_STATUS} from '../typings';
import {Colors} from 'src/constants';

interface Props {
  name: string;
  due_date: string;
  status: TASK_STATUS;
}

export const TaskItem: React.FC<Props> = ({name, due_date, status}) => {
  const [checked, setChecked] = useState(false);
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => setChecked(!checked)}
      activeOpacity={0.8}>
      <>
        <View style={styles.nameContainer}>
          <CheckBox boxType="square" style={styles.checkBox} value={checked} />
          {status === TASK_STATUS.DONE && (
            <MaterialIcons name="check-circle" size={20} color="green" />
          )}
          {status === TASK_STATUS.PROGRESS && (
            <EntypoIcons name="progress-one" size={20} color="blue" />
          )}
          {status === TASK_STATUS.PENDING && (
            <MaterialIcons name="pending" size={20} color="red" />
          )}
          <Text style={styles.nameText}>{name}</Text>
        </View>
        <Text style={styles.dateText}>{due_date}</Text>
      </>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 10,
    paddingVertical: 15,
    backgroundColor: Colors.darkWhite,
    borderColor: Colors.aliceBlue,
    borderWidth: 1,
    borderRadius: 3,
    width: Dimensions.get('screen').width - 20,
  },
  nameContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  nameText: {
    marginLeft: 5,
    color: Colors.darkPurple,
  },
  dateText: {
    color: Colors.darkBlue,
  },
  checkBox: {
    // width: 20,
    height: 20,
  },
});
