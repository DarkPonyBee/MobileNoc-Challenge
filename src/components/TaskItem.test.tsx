import React from 'react';
import {TaskItem} from './TaskItem';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {TASK_STATUS} from 'src/typings';

describe('TaskItem', () => {
  it('should render without issue', () => {
    const component = shallow(
      <TaskItem name="name" due_date="" status={TASK_STATUS.DONE} />,
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render task title properly', () => {
    const title = 'test_name';
    const component = shallow(
      <TaskItem name={title} due_date="" status={TASK_STATUS.DONE} />,
    );

    expect(component.find('Text').first().props().children).toBe(title);
  });

  it('should render date properly', () => {
    const dueDate = '22:00AM 12-1-2019';
    const component = shallow(
      <TaskItem name="" due_date={dueDate} status={TASK_STATUS.DONE} />,
    );
    expect(component.find('Text').last().props().children).toBe(dueDate);
  });

  it('should update checkbox properly', () => {
    const component = shallow(
      <TaskItem name="" due_date="" status={TASK_STATUS.DONE} />,
    );
    expect(component.find('View').children().first().props().value).toBe(false);
    component.props().onPress();
    expect(component.find('View').children().first().props().value).toBe(true);
  });

  it('should render icon properly', () => {
    const doneComponent = shallow(
      <TaskItem name="" due_date="" status={TASK_STATUS.DONE} />,
    );
    expect(doneComponent.find('Icon').props().name).toBe('check-circle');
    const progressComponent = shallow(
      <TaskItem name="" due_date="" status={TASK_STATUS.PROGRESS} />,
    );
    expect(progressComponent.find('Icon').props().name).toBe('progress-one');
    const pendingComponent = shallow(
      <TaskItem name="" due_date="" status={TASK_STATUS.PENDING} />,
    );
    expect(pendingComponent.find('Icon').props().name).toBe('pending');
  });
});

Enzyme.configure({adapter: new Adapter()});
