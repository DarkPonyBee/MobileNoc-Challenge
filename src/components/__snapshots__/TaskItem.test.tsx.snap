// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem should render without issue 1`] = `
<ForwardRef
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(50, 50, 50, 0.1)",
      "borderColor": "#F5FCFF",
      "borderRadius": 3,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 15,
      "width": 730,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <ForwardRef(CheckBoxWithRef)
      boxType="square"
      style={
        Object {
          "height": 20,
        }
      }
      value={false}
    />
    <Icon
      allowFontScaling={false}
      color="green"
      name="check-circle"
      size={20}
    />
    <Text
      style={
        Object {
          "color": "#3c345c",
          "marginLeft": 5,
        }
      }
    >
      name
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#124c81",
      }
    }
  />
</ForwardRef>
`;
